### PR TITLE
Prevent race on JdbcProjectionTest

### PR DESCRIPTION
**Pull request by ignasi35**
~There's a couple of flaky tests. Trying things out...~

References #264

### CODE REVIEW

File: .jvmopts-travis [hunk 1]
```diff
@@ -2,3 +2,4 @@
 
 -Dfile.encoding=UTF8
 -Dsbt.color=always
+-Dakka.actor.testkit.typed.timefactor=3
```
Comment (patriknw) [inline .jvmopts-travis:5]: time dilation is the wrong way of solving flaky tests, default 3 seconds should be enough, and for special initialization tasks there should be an explicit longer timeout

Comment (ignasi35) [reply .jvmopts-travis:5]: 👍🏼 Looking at alternatives.

**Comment by ignasi35** [comment]
I've repurposed the PR so instead of using a global time dilation I'm focusing on the flakiness of a single test.

File: akka-projection-jdbc/src/test/scala/akka/projection/jdbc/JdbcProjectionSpec.scala [hunk 2]
```diff
@@ -921,8 +922,13 @@ class JdbcProjectionSpec
...
       }
       withClue(s"check: last seen offset is 2L") {
-        val offset = offsetStore.readOffset[Long](projectionId).futureValue.value
-        offset shouldBe 2L
+        val configForFutureValue = implicitly[PatienceConfig]
+        val eventuallyTimeout = Timeout(configForFutureValue.timeout.scaledBy(3))
+        eventually(eventuallyTimeout) {
+          val offset = offsetStore.readOffset[Long](projectionId).futureValue.value
+          offset shouldBe 2L
+        }
```
Comment (ignasi35) [inline akka-projection-jdbc/src/test/scala/akka/projection/jdbc/JdbcProjectionSpec.scala:930]: This is overly complex. A simple `eventually{...}` wrap (without all the specific timeout math) will be fine.

File: akka-projection-jdbc/src/test/scala/akka/projection/jdbc/JdbcProjectionSpec.scala [hunk 3]
```diff
@@ -921,8 +921,10 @@ class JdbcProjectionSpec
         concatStr.text shouldBe "abc|def|ghi"
       }
       withClue(s"check: last seen offset is 2L") {
-        val offset = offsetStore.readOffset[Long](projectionId).futureValue.value
-        offset shouldBe 2L
+        eventually {
```
Comment (patriknw) [inline akka-projection-jdbc/src/test/scala/akka/projection/jdbc/JdbcProjectionSpec.scala:924]: That's right, it's not guaranteed that it has saved 2L before processing 3 and 4. There are probably many more of this, also in Cassandra and Slick tests.

**Comment by patriknw** [comment]
it still failed though:
```
<-- [akka.projection.jdbc.JdbcProjectionSpec: A JDBC at-least-once projection must restart from previous offset - handler throwing an exception, save after 2] End of log messages of test that [Failed(org.scalatest.exceptions.TestFailedDueToTimeoutException: check: last seen offset is 2L The code passed to eventually never returned normally. Attempted 27 times over 5.160632144999999 seconds. Last failure message: The Option on which value was invoked was not defined..)]
559[info] - must restart from previous offset - handler throwing an exception, save after 2 *** FAILED *** (5 seconds, 596 milliseconds)
560[info]   check: last seen offset is 2L The code passed to eventually never returned normally. Attempted 27 times over 5.160632144999999 seconds. Last failure message: The Option on which value was invoked was not defined.. (JdbcProjectionSpec.scala:924)
```

maybe the other fix in https://github.com/akka/akka-projection/pull/281 will help

**Comment by ignasi35** [comment]
> maybe the other fix in #281

Yeah, upon further inspection I also started suspecting multiple accesses on the same schema could be the culprit. Those are solved in #281 (along with random projection IDs...).

I think we can close this one for the time being.

File: akka-projection-jdbc/src/test/scala/akka/projection/jdbc/JdbcProjectionSpec.scala (hunk 3 above)
Comment (ignasi35) [reply akka-projection-jdbc/src/test/scala/akka/projection/jdbc/JdbcProjectionSpec.scala:924]: Yeah, my idea behind the refactor in #284 was to propagate this `eventually` to everywhere.

**Pull request closed without merging**